gerrit/changeset_test.go

```diff
@@ -0,0 +1,55 @@
+package gerrit
+
+import (
+	"testing"
+
+	goGerrit "github.com/andygrunwald/go-gerrit"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsAutosubmit(t *testing.T) {
+	emptyChangeInfo := &goGerrit.ChangeInfo{}
+	assert.Equal(t, false, MakeChangeset(emptyChangeInfo).IsAutosubmit(), "A changeset without the Autosubmit label present shouldn't be autosubmittable")
+
+	// +1
+	changeInfoWithAutosubmitLabelSet := &goGerrit.ChangeInfo{
+		Labels: map[string]goGerrit.LabelInfo{
+			"Autosubmit": {
+				Approved: goGerrit.AccountInfo{AccountID: 1},
+			},
+		},
+	}
+	assert.Equal(t, true, MakeChangeset(changeInfoWithAutosubmitLabelSet).IsAutosubmit(), "Autosubmit label set to +1 should be autosubmittable")
+
+	// some odd corner cases, can mostly happen due to broken label config
```

Comment:
I think it's wrong to consider these odd corner cases. People can have all sorts of logic implemented here that you may not anticipate.

Reply:
This mostly tests that values outside 0 and 1 (but those still commonly seen in gerrit) won't cause IsAutosubmit to return true. Why is it harmful to have that in a unit test?

We obviously can't exhaustively check for the whole range of int.

Reply:
It's not harmful to test of course, I'm just criticising the phrasing of the comment. There are cases like this that are not broken config is all :)

Reply:
Gotcha, I updated the comment. It now says

```
ensure some common label values don't trigger autosubmit. We only trigger on a "+1"
```

Which is more neutral.

Thanks for the review!